Add execname support

daemon1024 opened the pull request:
**Purpose of PR?**:

Based on discussion at https://kubearmor.slack.com/archives/C01F9V3SEHY/p1709480346589829

**Does this PR introduce a breaking change?**
NA

**If the changes in this PR are manually verified, list down the scenarios covered:**:
Added testcase

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_


**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

<!--

The PR title message must follow convention:
`<type>(<scope>): <subject>`.

Where: <br />
- `type` is to define what type of PR is this.
  Most common types are:
    - `feat`      - for new features, not a new feature for build script
    - `fix`       - for bug fixes or improvements, not a fix for build script
    - `chore`     - changes not related to production code
    - `docs`      - changes related to documentation
    - `style`     - formatting, missing semi colons, linting fix etc; no significant production code changes
    - `test`      - adding missing tests, refactoring tests; no production code change
    - `refactor`  - refactoring production code, eg. renaming a variable or function name, there should not be any significant production code changes

- `scope` is a single word that best describes where the changes fit.
    - feature(`monitor`,`enforcer`)
    - test(`tests`, `bdd`)
    - chore(`build`)
- `subject` is a single line brief description of the changes made in the pull request.

-->

daemon1024 committed:
enhance(bpflsm): match just dentry names to support execname policymatching

This will help us to create and enforce policies with
- path: apt
instead of
- path: /usr/bin/apt
- path: /bin/apt
and the likes

Signed-off-by: daemon1024 <barun1024@gmail.com>

daemon1024 committed:
feat(crd): support execname in Process spec

Signed-off-by: daemon1024 <barun1024@gmail.com>

daemon1024 committed:
feat(enforcer,feeder): parse execname into lsm rules and support policy matching on alerts

Signed-off-by: daemon1024 <barun1024@gmail.com>

daemon1024 committed:
test: add execname testcase

Signed-off-by: daemon1024 <barun1024@gmail.com>

daemon1024 commented on deployments/CRD/KubeArmorPolicy.yaml:
```diff
@@ -358,8 +361,6 @@ spec:
                           items:
                             type: string
                           type: array
-                      required:
-                      - path
```
Ideally we want either path/bin to be required. 
CRDs do tend to support it through `oneOf` concept but kubebuilder has no API to create that validation

daemon1024 committed:
fix(hsp/systemd): parse both matchpaths and matchdir together on hsp event

Signed-off-by: daemon1024 <barun1024@gmail.com>

nyrahul commented on KubeArmor/core/unorchestratedUpdates.go:
```diff
@@ -677,21 +677,20 @@ func (dm *KubeArmorDaemon) restoreKubeArmorPolicies() {
 
 				} else { // HostSecurityPolicy
-					var hostPolicy tp.HostSecurityPolicy
+					var hostPolicy tp.K8sKubeArmorHostPolicy
```
Is there a difference between k8s based host policy and VM based host policy?

daemon1024 replied:
So tp.K8sKubeArmorHostPolicy = Status Field + tp.HostSecurityPolicy

No difference in actual Policy Structure

We don't use the Status Field for VM.
We send gRPC events as K8s events as well so as they are handled in a streamlined manner and they don't need special handling.

nyrahul commented on KubeArmor/BPF/enforcer.bpf.c:
```diff
@@ -364,12 +377,11 @@ static inline int match_net_rules(int type, int protocol, u32 eventID) {
 
-  bpf_map_update_elem(&bufk, &one, z, BPF_ANY);  
-  p->path[0] = dnet ;
+  bpf_map_update_elem(&bufk, &one, z, BPF_ANY);
```
Can we add pre-commit hooks or a github action to fail if such whitespaces are added? We can have a GFS for this.

daemon1024 replied:
These were added by clang format. I will create an issue to have clang formatting on and verify it like we do for go-fmt.

nyrahul reviewed (dismissed):
LGTM. Few minor comments.

daemon1024 committed:
fix(hsp): missing policy name during restoration

Signed-off-by: daemon1024 <barun1024@gmail.com>

DelusionalOptimist commented on KubeArmor/enforcer/appArmorProfile.go:
Need to add handling for `execname` rules in AppArmor profiles for host.

daemon1024 committed:
feat(hsp): support execname on apparmor hsp

Signed-off-by: daemon1024 <barun1024@gmail.com>

daemon1024 committed:
feat(snitch): use containerd socket in docker runtime if available

docker events seem to be causing flakey behavior. docker under the hood does use containerd runtime. this commit checks if containerd socket is availble in docker runtime and uses it.

Signed-off-by: daemon1024 <barun1024@gmail.com>

DelusionalOptimist reviewed (dismissed):
LGTM.

DelusionalOptimist committed:
chore: compile object files again on Linux 5.15

Signed-off-by: Rudraksh Pareek <rudraksh@accuknox.com>

daemon1024 committed:
hotfix(ci): disabling docker from runtime

no idea why it's so flakey after k3s update

Signed-off-by: daemon1024 <barun1024@gmail.com>
